src/reset.cpp

```diff
@@ -30,6 +30,7 @@ namespace libcellml {
 struct Reset::ResetImpl
 {
     int mOrder = 0; /**< An integer for determining relative order.*/
+    char padding[4];
```

Comment:
Not at all keen on these padding items in the structure what exactly is the issue that clang is having here.